pkg/mimirpb/compat.go

```diff
@@ -156,9 +156,14 @@ func FromExemplarProtosToExemplars(es []Exemplar) []exemplar.Exemplar {
 	return result
 }
 
-// FromPointsToSamples casts []promql.Point to []Sample. It uses unsafe.
+// FromPointsToSamples converts []promql.Point to []Sample.
 func FromPointsToSamples(points []promql.Point) []Sample {
```

Comment:
The Point could have had a histograms here. We should either fix it, or add a TODO saying if there are histograms, capture that properly. Something like `FromPointsToSamplesAndHistograms(points) ([]Sample, []Histogram)`